modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -326,6 +326,10 @@ def _deploy_ray_func(
     if isinstance(result, pandas.DataFrame):
         return result, len(result), len(result.columns), ip
     elif all(isinstance(r, pandas.DataFrame) for r in result):
-        return [i for r in result for i in [r, len(r), len(r.columns), ip]]
+        for r in result:
+            for item in [r, len(r), len(r.columns), ip]:
+                yield item
```

Comment:
I think that [the tip to use generators](https://docs.ray.io/en/latest/ray-core/patterns/generators.html) only helps in cases where each iteration of the yielding for-loop actually creates some objects/allocates new memory. In this for loop we already have all the objects computed (dfs in `result`) and the memory for <i>all the results</i> was already allocated. 

Can we instead use generators in [`split_result_of_axis_func_pandas`](https://github.com/modin-project/modin/blob/21d10ea8f6c91a83787d50f09528cbc2e94f2884/modin/core/dataframe/pandas/partitioning/axis_partition.py#L454) function that actually creates a list of resulting dataframes?

Reply:
It seems we should use generators here and in `split_result_of_axis_func_pandas` too.